scheduler/system_util.go

```diff
@@ -98,16 +100,6 @@ func diffSystemAllocsForNode(
 			continue
 		}
 
-		// If we are a sysbatch job and terminal, ignore (or stop?) the alloc
-		if job.Type == structs.JobTypeSysBatch && exist.TerminalStatus() {
-			result.ignore = append(result.ignore, allocTuple{
-				Name:      name,
-				TaskGroup: tg,
-				Alloc:     exist,
-			})
-			continue
-		}
```

Comment:
Note to reviewers: this is dead code, confirmed by test coverage from the original tests plus the refactored ones. Only non-terminal allocs are passed in the list we're iterating over here for the `exist` block.

Reply:
Is the first condition in the block above also unnecessary then?

```
if !exist.TerminalStatus() && exist.DesiredTransition.ShouldMigrate() {
```

Reply:
Looks like it!